OpenRA.Game/ModData.cs

```diff
@@ -144,11 +149,15 @@ public Map PrepareMap(string uid)
 			using (new Support.PerfTimer("Map"))
 				map = new Map(this, MapCache[uid].Package);
 
-			// Reinitialize all our assets
+			// Reinitialize assets.
 			InitializeLoaders(map);
-			map.Sequences.LoadSprites();
+			if (!sequencesLoaded)
+			{
+				map.Sequences.LoadSprites();
+				sequencesLoaded = true;
```

Comment:
`sequencesLoaded` feel somewhat dirty. 

Wouldn't have been cleaner to maintain `lastSequences`/`sequencesLoadedCache` loaded and then reuse that at the moment you want to load the new set if they match.

But perhaps this is about `GetSequences` being called prior to `PrepareMap`.